### PR TITLE
fix(Core/Unit): potential crash + extract duplicated code

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
## Issues Addressed:
- Might close https://github.com/azerothcore/azerothcore-wotlk/issues/5272 

## Tests Performed:

- Not tested

## How to Test the Changes:

- not really sure what to test this
- not sure how to reproduce the original crash. @n4ndo should test this PR.

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -11351,6 +11333,40 @@ uint32 Unit::SpellDamageBonusTaken(Unit* caster, SpellInfo const* spellProto, ui
     return uint32(std::max(tmpDamage, 0.0f));
 }
 
+float Unit::processDummyAuras(float TakenTotalMod) const {
```
Comment (Kitzunu) [inline src/server/game/Entities/Unit/Unit.cpp:11336]: { newline

File: src/server/game/Entities/Unit/Unit.cpp [hunk 2]
```diff
@@ -11351,6 +11333,40 @@ uint32 Unit::SpellDamageBonusTaken(Unit* caster, SpellInfo const* spellProto, ui
...
+    // note: old code coming from TC, just extracted here to remove the code duplication + solve potential crash
+    // see: https://github.com/TrinityCore/TrinityCore/commit/c85710e148d75450baedf6632b9ca6fd40b4148e
```
Comment (Kitzunu) [inline src/server/game/Entities/Unit/Unit.cpp:11338]: Unnecessary

Comment (FrancescoBorzi) [reply src/server/game/Entities/Unit/Unit.cpp:11338]: it is necessary so people don't assume this code is newly added when checking the code with annotations open (git blame)

File: src/server/game/Entities/Unit/Unit.cpp (hunk 1 above)
Comment (FrancescoBorzi) [reply src/server/game/Entities/Unit/Unit.cpp:11336]: thanks, I'll fix it

**Comment by mpfans** [comment]
Recently, my server will crash every six hours or so. I will test this pr

**Comment by Kitzunu** [comment]
@mpfans could you attach a crashlog from RelWithDebInfo or Debug?

**Comment by mpfans** [comment]
> @mpfans could you attach a crashlog from RelWithDebInfo or Debug?

no..I don't know how to get crash log

**Comment by FrancescoBorzi** [comment]
@mpfans I think if this PR does not break anything, we should merge it. But I haven't tested it in-game myself.

**Comment by Kitzunu** [comment]
> > @mpfans could you attach a crashlog from RelWithDebInfo or Debug?
> 
> no..I don't know how to get crash log

In the Crash folder, but if you build in "Release" it doesn't matter

**Comment by mpfans** [comment]
> But I haven't tested it in-game myself.

I'm already testing. Please wait a few days. I'll give you feedback

**Comment by mpfans** [comment]
this PR can merge it.. tested

**Pull request merged**